pkg/compact/downsample/downsample.go

```diff
@@ -617,15 +618,21 @@ func (it *ApplyCounterResetsSeriesIterator) Next() chunkenc.ValueType {
 		if it.i >= len(it.chks) {
 			return chunkenc.ValNone
 		}
-		if it.chks[it.i].Next() == chunkenc.ValNone {
+		it.lastValType = it.chks[it.i].Next()
```

Comment:
Is this change related to this pr? Downsample/Compact side change seems not related to QFE and UI.

Reply:
You are right. It's not directly related but it's a fix so that `rate()` works for native histograms: https://github.com/thanos-io/thanos/blob/fdeea3917591fc363a329cbe23af37c6fff0b5f0/pkg/query/iter.go#L131

I will add this information to the first comment.